benchmark_utils/backbones_architecture.py

```diff
@@ -98,5 +96,72 @@ def __init__(self, n_chans, n_classes, input_window_samples):
     def forward(self, x, sample_weight=None):
         x = self.feature_layer(x)
         x = self.final_layer(x)
-    
+
         return x
+
+
+class GradientReversalLayer(Function):
+    """Leaves the input unchanged during forward propagation
+    and reverses the gradient by multiplying it by a
+    negative scalar during the backpropagation.
+    """
+
+    @staticmethod
+    def forward(ctx, x, alpha, sample_weight=None):
+        """XXX add docstring here."""
+        ctx.alpha = alpha
+
+        return x.view_as(x)
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        """XXX add docstring here."""
+        output = grad_output.neg() * ctx.alpha
+        return output, None
+
+
+class DomainClassifier(nn.Module):
+    """Classifier Architecture from DANN paper [15]_.
+
+    Parameters
+    ----------
+    num_features : int
+        Size of the input, e.g size of the last layer of
+        the feature extractor
+    n_classes : int, default=1
+        Number of classes
+
+    References
+    ----------
+    .. [15]  Yaroslav Ganin et. al. Domain-Adversarial Training
+            of Neural Networks  In Journal of Machine Learning
+            Research, 2016.
+    """
+
+    def __init__(self, num_features, hidden_size=1024, n_classes=1, alpha=1):
+        super().__init__()
+        self.classifier = nn.Sequential(
+            nn.Linear(num_features, hidden_size),
+            nn.ReLU(),
+            nn.Dropout(0.5),
+            nn.Linear(hidden_size, hidden_size),
+            nn.ReLU(),
+            nn.Dropout(0.5),
+            nn.Linear(hidden_size, n_classes),
+            nn.Sigmoid(),
```

Comment:
Why a sigmoid? e.g. `ShallowConvNet` returns logits.

Reply:
Its the domain classifier. It does Binary classification between the Target and Source domain. That's why using a sigmoid makes sense

Reply:
Ok, I just saw we use `torch.nn.BCELoss()` in skada so the pr looks good to me!